design/src/rfcs/rfc0012_body_callback_apis.md

```diff
@@ -0,0 +1,259 @@
+RFC: Callback APIs for `ByteStream` and `SdkBody`
+=================================================
+
+> Status: RFC
+
+Adding a callback APIs to `ByteStream` and `SdkBody` will enable developers using the SDK to implement things like checksum validations and 'read progress' callbacks.
```

Comment:
Are the validations on requests or responses? Is the `ReadCallback` intended for both, or just for requests?

Reply:
It can work for both, it just gets used slightly differently each time. I'll clarify that more.